### ISSUE TITLE
Get all pages

### ISSUE THREAD

**Issue by mvaldes14** [jamalex/notion-py#43]
Is there a way to pass a token and get all pages for a user?
most of the methods require you to pass a url to get the ID of the blocks and then interact with them but was wondering if there could be a higher function to get everything in a user workspace.

been trying to understand whats the real endpoint the api hits and from what i can see its something like notion.so/api/v3 but then what else is passed down to it? could that be something on the wiki page for people to test out things?

**Comment by iansinnott** [comment]
I think there must be a better way to do this, but the following code works for me. The key starting point is `client.current_space.pages`:

```py
def get_all_notion_pages():
    def get_children(block):
        result = []
        for child in block.children:
            if child.type == 'page':
                result.append(child)
                result += get_children(child)
        return result

    pages = []

    for page_id in client.current_space.pages:
        block = client.get_block(page_id)
        if block.type == 'page':
            pages.append(block)
            pages += get_children(block)

    return pages

```

**Comment by jamalex** [comment]
Thanks for the issue!

Sort of a duplicate of https://github.com/jamalex/notion-py/issues/50 -- and has been addressed in #87, which will be in an upcoming release.

**Comment by mvaldes14** [comment]
much appreciated!
